rls/src/main/java/io/grpc/rls/RlsLoadBalancer.java

```diff
@@ -95,7 +95,7 @@ public void handleNameResolutionError(final Status error) {
     class ErrorPicker extends SubchannelPicker {
       @Override
       public PickResult pickSubchannel(PickSubchannelArgs args) {
-        return PickResult.withError(error);
+        return PickResult.withError(CachingRlsLbClient.convertServerStatus(error, args));
```

Comment:
This looks wrong. handleNameResolutionError is coming from the name resolver, but the error message will say the error came from RLS server. Right now most policies are propagating `error` without transformation, relying on the NR to produce a good error message.